### ISSUE TITLE
[Feature Request] Freezing Rows and Columns

### ISSUE THREAD

**Issue by dmorrison42** [noahmorrison/genodf#1]
I need the ability to freeze the rows (and columns?) for display and printing.

This appears to be two separate settings, both of which are saved to file, so may be part of the scope of this project.

https://help.libreoffice.org/Calc/Freezing_Rows_or_Columns_as_Headers
https://help.libreoffice.org/Calc/Printing_Rows_or_Columns_on_Every_Page

**Comment by dmorrison42** [comment]
The most obvious changes are within the content.xml file.

Print Lock (Repeating)

```diff
$ diff -u no\ lock\ content.xml print\ lock\ content.xml -b
--- "no lock content.xml"       2018-03-30 09:38:01.105259000 -0400
+++ "print lock content.xml"    2018-03-30 09:42:34.518259000 -0400
@@ -78,11 +78,13 @@
             <table:calculation-settings table:automatic-find-labels="false" table:use-regular-expressions="false" table:use-wildcards="true"/>
             <table:table table:name="Sheet1" table:style-name="ta1">
                 <table:table-column table:style-name="co1" table:default-cell-style-name="Default"/>
+                <table:table-header-rows>
                 <table:table-row table:style-name="ro1">
                     <table:table-cell office:value-type="string" calcext:value-type="string">
                         <text:p>test</text:p>
                     </table:table-cell>
                 </table:table-row>
+                </table:table-header-rows>
                 <table:table-row table:style-name="ro1">
                     <table:table-cell office:value-type="float" office:value="1" calcext:value-type="float">
                         <text:p>1</text:p>
```

Visual Lock (Freezing)

```diff
--- "no lock content.xml"       2018-03-30 09:38:01.105259000 -0400
+++ "visual lock content.xml"   2018-03-30 09:41:37.218259000 -0400
@@ -77,6 +77,7 @@
         <office:spreadsheet>
             <table:calculation-settings table:automatic-find-labels="false" table:use-regular-expressions="false" table:use-wildcards="true"/>
             <table:table table:name="Sheet1" table:style-name="ta1">
+                <office:forms form:automatic-focus="false" form:apply-design-mode="false"/>
                 <table:table-column table:style-name="co1" table:default-cell-style-name="Default"/>
                 <table:table-row table:style-name="ro1">
                     <table:table-cell office:value-type="string" calcext:value-type="string">
```

**Comment by noahmorrison** [comment]
Closed with ad45f8476c6ef32d9ebd4b60b31e5d96b4ec9bb3

Use sheet.PrintingHeader for repeating rows/columns on every printed page

Use sheet.VisualHeader for freezing rows/columns visually
